### PR TITLE
Okta auth 2

### LINKED ISSUE #301: OKTA Integration
As a CARTS user I need to be able login using my EUA ID and password so that I can access the application.

AC:

- [ ] Successfully login to CARTS

- [ ] OKTA passes back to application user job code

### PR DESCRIPTION
**Pull request by sethsacher**

### CODE REVIEW

**Comment by sethsacher** [comment]
Replaces https://github.com/18F/cms-carts-seds/pull/334

**Review by tadhg-ohiggins — APPROVED** [review]

File: frontend/react/src/App.js [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-
-  let VisibleFooter =
-    window.location.pathname.split("/")[1] === "reports" ? null : <Footer />;
   return (
-    <div className="App" data-test="component-app">
-      {VisibleHeader}
-      <Routes />
-      {VisibleFooter}
-    </div>
+    <Router>
```
Comment (cannandev) [inline frontend/react/src/App.js:11]: 😊  glad you're using Router

**Review by cannandev — CHANGES_REQUESTED** [review]

File: frontend/react/src/Home.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import "font-awesome/css/font-awesome.min.css";
+import "./App.scss";
+import Routes from "./reactRouter";
+import Header from "./components/layout/Header";
+import Footer from "./components/layout/Footer";
+
+function Home() {
+  let VisibleHeader =
+    window.location.pathname.split("/")[1] === "reports" ? null : <Header />;
```
[the remark this answers is not in the document]
Comment (sethsacher) [reply frontend/react/src/Home.js:10]: I didn't add this - it was just moved over from App.js into this new Home.js component. @jgillis1 should review in a separate PR.

File: frontend/react/src/components/layout/Logout.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+import config from '../../auth-config';
+
+const issuer = 'https://${yourOktaDomain}/oauth2/default';
```
[the remark this answers is not in the document]
Comment (sethsacher) [reply frontend/react/src/components/layout/Logout.js:5]: Whoops this shouldn't be there at all, it's carry over from a template I used. I'll remove it.

File: frontend/react/src/auth-config.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export default {
+    oidc: {
+        clientId: '0oa4juv4poiQ6nDB6297',
+        issuer: 'https://test.idp.idm.cms.gov/oauth2/aus4itu0feyg3RJTK297',
+        redirectUri: 'http://localhost:81/implicit/callback',
+        scopes: ['openid', 'profile', 'email'],
+        pkce: true
+    },
+    callback: '/implicit/callback'
+  };
```
Comment (sethsacher) [inline frontend/react/src/auth-config.js:10]: @cannandev do you suggest a new line here as well?

**Comment by sethsacher** [comment]
Okta-based authentication wrapping the entire CARTS React frontend. If a user is not authenticated, they will be directed to the default CMS Okta login screen. After providing valid credentials, the user is directed to the CARTS homepage. This is currently only implemented for local development, and only for the frontend. Additional checks should be implemented on the Django backend to ensure an unauthenticated user cannot access an API endpoint.

Logout was added as well. This is configured to route the user back to root, which will then redirect to the default CMS Okta login screen per the auth workflow.

To authenticate, a user must exist in the Okta application "MACPRO SPA". I have admin privileges right now and can add users for you, and you will likely want to request admin privileges from the Okta team to manage this moving forward. This integration applies authentication to a local build. You'll either want to follow up with code to apply auth only to certain environments, or add users to the MACPRO SPA app in order to login appropriately. Let me know if you have any questions about how to configure additional environments (this will require adding the new environment URL to Okta, and possibly creating the MACPRO SPA Okta app in higher Okta environments).

We plan to generalize the auth code in a reusable component across MACPRO Apps. I'll follow up with the team at a later date if/when that becomes a reality.

File: frontend/react/src/Home.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgillis1) [reply frontend/react/src/Home.js:10]: I agree. We can address this through another Issue/PR.

File: frontend/react/src/auth-config.js (hunk 4 above)
Comment (cannandev) [reply frontend/react/src/auth-config.js:10]: yes. thanks!

**Review by cannandev — APPROVED** [review]

**Pull request merged**